scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -337,6 +339,67 @@ object GetSettings extends EsScript {
     sys.exit(1)
   }
 }
+object DownloadAllEsIds extends EsScript {
+
+  def run(esUrl: String, extraArgs: List[String]) {
+
+    println(s"Getting all ids from $esUrl")
+
+    object Client extends EsClient(esUrl) {
+
+      def writeAllEsIdsToFile(filename: String) = {
+
+        val file = new File(filename)
+
+        val writer = new PrintWriter(file)
```

Comment:
reading through the class docs, I don't think this will automatically flush the buffer, so I think it's possible you could end up with the report entirely buffered in memory, which might cause an OOM on large indices - though idk if our PROD is big enough to cause that
Might be worth explicitly `.flush()`ing every 100k lines or something, just to prevent huge buildups?

Reply:
alternatively you could construct with a `FileOutputStream`, and then you can set automatic flushing (though that'll flush on each println, which I think might be excessive) https://docs.oracle.com/javase/8/docs/api/java/io/PrintWriter.html#PrintWriter-java.io.OutputStream-boolean-

Reply:
good thought, but it seems to flush fairly promptly (as it does the same in https://github.com/guardian/grid/pull/4115) and it didn't OOM when running against PROD